src/components/new-repository/NewRepository/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { Container, FormControl, FormLabel, Input, Select, Stack, StackProps, Textarea } from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import { FC } from 'react';
+import { useFormContext } from 'react-hook-form';
+import useSWR from 'swr';
+import { RepoFields, RepoFormValues } from '../CreateRepositoryFormLayout/utils';
+
+interface INewRepositoryProps {}
+
+export const NewRepository: FC<StackProps> = (props) => {
+	const { register } = useFormContext<RepoFormValues>();
+	const router = useRouter();
+	const { data: repos } = useSWR('/api/github/user/repos', {
+		onError: (error) => {
+			if (error.code === '403') {
+				router.push('/repos/github');
+			}
+		},
+	});
+	if (!repos) {
+		return null;
```

Comment:
Maybe show a spinner here?

```suggestion
		return <Center><Spinner /></Center>;
```